### PR TITLE
add cni plugin config file max num config

### PR DESCRIPTION
**Pull request by kuramal**
Make cni plugin config file num configurable， let containerd can make mutil network card

#1154

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -71,6 +71,8 @@ type CniConfig struct {
 	NetworkPluginBinDir string `toml:"bin_dir" json:"binDir"`
 	// NetworkPluginConfDir is the directory in which the admin places a CNI conf.
 	NetworkPluginConfDir string `toml:"conf_dir" json:"confDir"`
+	// NetworkPluginMaxConfNum is the max num of plugin file will load
```
Comment (Random-Liu) [inline pkg/config/config.go]: s/max num of plugin file/max number of plugin config file

Comment (mikebrow) [reply pkg/config/config.go]: Can we explain what this number is and how it is used?

Comment (mikebrow) [reply pkg/config/config.go]: Can you link me to the value is used pls?

File: pkg/config/config.go [hunk 2]
```diff
@@ -192,6 +194,7 @@ func DefaultConfig() PluginConfig {
 		CniConfig: CniConfig{
 			NetworkPluginBinDir:       "/opt/cni/bin",
 			NetworkPluginConfDir:      "/etc/cni/net.d",
+			NetworkPluginMaxConfNum:   1,
```
Comment (mikebrow) [inline pkg/config/config.go]: hmm.. should we be loading this from some vendor constant over in cni or go-cni? Sort of goes with the above question, where is this going to be used. Are there any test cases?

File: pkg/config/config.go (hunk 1 above)
Comment (Random-Liu) [reply pkg/config/config.go]: Ha, yeah we should update docs/config.md.

Comment (kuramal) [reply pkg/config/config.go]: @Random-Liu Yeah, that's right. 
@mikebrow In our case, we add a second NIC of the pod, and set a LAN ip for it.The second NIC is Optional，and we control the ips for every deployment stable.

Comment (mikebrow) [reply pkg/config/config.go]: Ok, interesting use case... 

I don't see code in cni proper for using this value.. but I may have missed it? Is this for a custom plugin or a to be dropped pr for cni? Is this value just a pass through value? Is there a mockup of this pattern somewhere? Inquiring minds want to know :-)

Comment (Random-Liu) [reply pkg/config/config.go]: The value is used in `go-cni` library. :)

Comment (Random-Liu) [reply pkg/config/config.go]: @kuramal Can you update https://github.com/containerd/cri/blob/master/docs/config.md, and include some explanation of the option, like https://github.com/containerd/cri/pull/1155#discussion_r291023199?

**Comment by Random-Liu** [comment]
Waiting for document change, otherwise LGTM

File: pkg/config/config.go (hunk 1 above)
Comment (mikebrow) [reply pkg/config/config.go]: ok found it in go-cni/opts.go 

// loadFromConfDir detects network config files from the
// configured cni config directory and load them. max is
// the maximum network config to load (max i<= 0 means no limit).

File: pkg/config/config.go (hunk 2 above)
Comment (mikebrow) [reply pkg/config/config.go]: this replaces the default .. which is by default 1 so no observable behavior change here.. just odd that we are not vendoring the default from go-cni

File: pkg/config/config.go [hunk 3]
```diff
@@ -71,6 +71,8 @@ type CniConfig struct {
 	NetworkPluginBinDir string `toml:"bin_dir" json:"binDir"`
 	// NetworkPluginConfDir is the directory in which the admin places a CNI conf.
 	NetworkPluginConfDir string `toml:"conf_dir" json:"confDir"`
+	// NetworkPluginMaxConfNum is the max number of plugin config file will load
```
Comment (mikebrow) [inline pkg/config/config.go]: // NetworkPluginMaxConfNum (default = 1) is the max number of plugin config files that
// will be loaded from the cni config directory. Set the value to 0 to load all config files (no arbitrary limit).

Comment (mikebrow) [reply pkg/config/config.go]: And say something like that in the config.md.

File: pkg/config/config.go (hunk 1 above)
Comment (kuramal) [reply pkg/config/config.go]: @Random-Liu  Done

File: docs/config.md [hunk 4]
```diff
@@ -149,6 +149,12 @@ The explanation and default value of each configuration item are as follows:
     # conf_dir is the directory in which the admin places a CNI conf.
     conf_dir = "/etc/cni/net.d"
 
+    # max_conf_num is the max number of CNI plugin config file will load
+    # By default, the containerd will load only 1 CRI plugin. If you want load multiple CRI plugins,
+    # set max_conf_num > 1. You should know exactly each CRI plugin do, whether they are compatible.
+    # max i<= 0 means no limit.
```
Comment (mikebrow) [inline docs/config.md]: typo nits:
    # max_conf_num specifies the maximum number of CNI plugin config files that will be loaded from the CNI config directory.    
    # By default, containerd will load only 1 CNI plugin config file. If you want to load multiple CNI plugin config files set max_conf_num to the number desired or 0, as 0 means no limit (load all CNI plugin config files from the CNI config directory).

**Comment by mikebrow** [comment]
@Random-Liu I'm fine with merging this and doing a follow up for the text descriptions. The code is fine!

**Comment by Random-Liu** [comment]
/lgtm

@mikebrow Can you send a doc update? Thanks!

**Comment by mikebrow** [comment]
sure first thing in the morn...

**Pull request merged**